### PR TITLE
Backport/do not allow duplicate role names

### PR DESCRIPTION
**Pull request by superkhau**
Added jshint ignore line to make pass linting.

### CODE REVIEW

File: test/role.test.js [hunk 1]
```diff
@@ -93,11 +90,25 @@ describe('role model', function() {
...
-  it('should automatically generate role id', function() {
+  it('should not allow duplicate role name', function(done) {
+    Role.create({ name: 'userRole' }, function(err, role) {
+      if (err) return done(err);
 
-    User.create({name: 'Raymond', email: 'x@y.com', password: 'foobar'}, function(err, user) {
-      // console.log('User: ', user.id);
-      Role.create({name: 'userRole'}, function(err, role) {
+      Role.create({ name: 'userRole' }, function(err, role) {
+        expect(err).to.exist; // jshint ignore:line
```
Comment (superkhau) [inline test/role.test.js]: @bajtos Do you know if there is a better way to fix this (besides `...to.exist()`).

File: test/role.test.js [hunk 2]
```diff
@@ -93,11 +90,25 @@ describe('role model', function() {
...
-  it('should automatically generate role id', function() {
+  it('should not allow duplicate role name', function(done) {
+    Role.create({ name: 'userRole' }, function(err, role) {
+      if (err) return done(err);
 
-    User.create({name: 'Raymond', email: 'x@y.com', password: 'foobar'}, function(err, user) {
-      // console.log('User: ', user.id);
-      Role.create({name: 'userRole'}, function(err, role) {
+      Role.create({ name: 'userRole' }, function(err, role) {
+        expect(err).to.exist; //jshint ignore:line
```
Comment (superkhau) [inline test/role.test.js]: @bajtos Do you know if there is a better way to fix this (besides ...to.exist())?

**Comment by superkhau** [comment]
@bajtos Review please.

File: test/role.test.js (hunk 2 above)
Comment (bajtos) [inline test/role.test.js]: That's fine with me, just squash the commit adding this line with the one you are fixing.

I think we should fix our eslint-config-loopback to reject lines like `expect(err).to.exist` too.

Comment (bajtos) [inline test/role.test.js]: > I think we should fix our eslint-config-loopback to reject lines like `expect(err).to.exist` too.

https://github.com/strongloop/eslint-config-loopback/pull/6

**Pull request merged**